### PR TITLE
Add CONTRIBUTING guidelines - Closes #367

### LINKED ISSUE #367: Add CONTRIBUTING file to repo root
So people know how to get started contributing to lisk-js

### PR DESCRIPTION
**Pull request by Tosch110**
Add CONTRIBUTING guidelines, CODE_OF_CONDUCT and ISSUE_TEMPLATE

Closes #367

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+In case you've never submitted a pull request (PR) via GitHub before, please read [this short tutorial](https://help.github.com/articles/creating-a-pull-request). If you've submitted a PR before, there should be nothing surprising about our procedures for Lisk.
+
+*Before* submitting a pull request, please make sure the following is done:
+
+1. Fork the repo and create your branch from `master`.
+2. If you've added code that should be tested, add tests!
+3. Ensure the test suite passes (`npm test`).
+4. Submit a pull request via GitHub.
+5. Check that Jenkins CI tests pass (pull request turns green).
```
Comment (Isabello) [inline CONTRIBUTING.md]: untrusted users cannot run jenkins

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    * :green_heart: `:green_heart:` when fixing the CI build
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :shirt: `:shirt:` when removing linter warnings
+
+### JavaScript Styleguide
+
+All JavaScript must adhere to [JavaScript Standard Style](http://standardjs.com/).
```
Comment (karmacoma) [inline CONTRIBUTING.md]: How does https://www.npmjs.com/package/eslint-config-lisk-base factor into this?

Comment (willclarktech) [reply CONTRIBUTING.md]: We currently don't use JavaScript Standard Style.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -144,6 +144,6 @@ Enhancement suggestions are tracked as [GitHub issues](https://guides.github.com
 
 ### JavaScript Styleguide
 
-All JavaScript must adhere to [JavaScript Standard Style](http://standardjs.com/).
+Our JavaScript style is expending the [airbnb](https://github.com/airbnb/javascript) style. You can get more details here: https://github.com/LiskHQ/eslint-config-lisk-base
 
-This Contributing guidelines was inspired by and is based on Atom's contribution guidelines and modified for purposes of this repository: https://github.com/atom/atom/blob/master/CONTRIBUTING.md
\ No newline at end of file
+This Contributing guidelines was inspired by and is based on Atom's contribution guidelines and modified for purposes of this repository: https://github.com/atom/atom/blob/master/CONTRIBUTING.md
```
Comment (willclarktech) [inline CONTRIBUTING.md]: `These contribution guidelines were inspired by and are based on Atom's contribution guidelines. They were modified for the purposes of this repository.`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+
+## I don't want to read this whole thing I just have a question!!!
+
+Lisk is an open-source decentralized project, there are many ways and platforms to get help. These are some of them
+
+* [Discuss and Ask on Reddit](https://www.reddit.com/r/Lisk/)
+* [Lisk FAQ](https://docs.lisk.io/docs/faq)
+
+If chat is more your speed, you can join the LiskHQ Gitter:
```
Comment (Isabello) [inline CONTRIBUTING.md]: If you prefer to chat with LiskHQ directly

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+* **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+* If the problem is related to working with files (e.g. opening and editing files), **does the problem happen for all files and projects or only some?** Does the problem happen only when working with local or remote files (e.g. on network drives), with files of a specific type (e.g. only JavaScript or Python files), with large files or files with very long lines, or with files in a specific encoding? Is there anything else special about the files you are using?
+
+Include details about your configuration and environment:
+
+* **Which version of Lisk / Lisk-JS are you using?**
+* **What's the name and version of the OS you're using**?
+* **Are you running Lisk-Js or Lisk in a virtual machine?** If so, which VM software are you using and which operating systems and versions are used for the host and the guest?
```
Comment (Isabello) [inline CONTRIBUTING.md]: Pick a standard for lisk-js, Lisk-JS or Lisk-Js

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+* **Are you using [local configuration files](https://docs.lisk.io/v1.2.1/docs/api/)** `config.json`? If so, provide the contents of those files, preferably in a [code block](https://help.github.com/articles/markdown-basics/#multiple-lines) or with a link to a [gist](https://gist.github.com/).
```
Comment (Isabello) [inline CONTRIBUTING.md]: Use the main published revision: https://docs.lisk.io/docs

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+First off, thanks for taking the time to contribute! :raised_hands:
+
+The following is a set of guidelines for contributing to Lisk-JS, which are hosted in the [LiskHQ Organization](https://github.com/LiskHQ) on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+#### Table Of Contents
+
+[Code of Conduct](#code-of-conduct)
+
+[I don't want to read this whole thing, I just have a question!!!](#i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
```
Comment (LindsaysDarkAlley) [inline CONTRIBUTING.md]: Help! I don't want to read this whole thing, I just have one question. :mag_right:

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+If you prefer to chat with LiskHQ and other developers directly::
+
+* [Join the LiskHQ Gitter](https://gitter.im/LiskHQ/lisk)
+* Even though Gitter is a chat service, sometimes it takes several hours for community members to respond &mdash; please be patient!
+
+## How Can I Contribute?
+
+LiskHQ uses GitHub as its sole source of truth. Everything happens here. Lightcurve employees who contribute to Lisk are expected to do so in the same way as everyone else. In other words, this document applies equally to all contributors.
+
+### `master` is unsafe
```
Comment (Isabello) [inline CONTRIBUTING.md]: master or development?

Comment (Tosch110) [reply CONTRIBUTING.md]: Both actually. `development`  is changing more often than `master` but we will also update `master` with every release.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+Provide more context by answering these questions:
+
+* **Did the problem start happening recently** (e.g. after updating to a new version of Lisk-JS, Lisk or any other repository) or was this always a problem?
+* If the problem started happening recently, **can you reproduce the problem in an older version of Lisk-JS?** What's the most recent version in which the problem doesn't happen? You can download older versions of Lisk-JS from [the releases page](https://github.com/LiskHQ/lisk-js/releases).
+* **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+* If the problem is related to working with files (e.g. opening and editing files), **does the problem happen for all files and projects or only some?** Does the problem happen only when working with local or remote files (e.g. on network drives), with files of a specific type (e.g. only JavaScript or Python files), with large files or files with very long lines, or with files in a specific encoding? Is there anything else special about the files you are using?
```
Comment (Tschakki) [inline CONTRIBUTING.md]: this passage can be deleted

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+* **Are you using [local configuration files](https://docs.lisk.io/docs)** `config.json`? If so, provide the contents of those files, preferably in a [code block](https://help.github.com/articles/markdown-basics/#multiple-lines) or with a link to a [gist](https://gist.github.com/).
+
+### Suggesting Enhancements
+
+This section guides you through submitting an enhancement suggestion for Lisk-JS, including completely new features and minor improvements to existing functionality. Following these guidelines helps maintainers and the community understand your suggestion :pencil: and find related suggestions :mag_right:.
+
+Before creating enhancement suggestions, please check [this list](#before-submitting-an-enhancement-suggestion) as you might find out that you don't need to create one. When you are creating an enhancement suggestion, please [include as many details as possible](#how-do-i-submit-a-good-enhancement-suggestion). Fill in [the template](ISSUE_TEMPLATE.md), including the steps that you imagine you would take if the feature you're requesting existed.
```
Comment (Tschakki) [inline CONTRIBUTING.md]: link to this list does not work

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+* **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+* **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+* **Describe the current behavior** and **explain which behavior you expected to see instead** and why.
+* **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of Lisk-JS which the suggestion is related to. You can use [this tool](http://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux.
+* **Explain why this enhancement would be useful** to most Lisk and Lisk-JS users.
+* **List some other text editors or applications where this enhancement exists.**
+* **Specify which version of Lisk and Lisk-JS you're using.**
+* **Specify the name and version of the OS you're using.**
+
+### Pull Requests
```
Comment (Tschakki) [inline CONTRIBUTING.md]: Pull Request are explained twice. Should be merged or deleted

File: ISSUE_TEMPLATE.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+**Expected behavior:** [What you expect to happen]
+
+**Actual behavior:** [What actually happens]
+
+**Reproduces how often:** [What percentage of the time does it reproduce?]
+
+### Versions
+
+Please provide the version you are working with. Also, please include the OS and what version of the OS you're running.
+
```
Comment (willclarktech) [inline ISSUE_TEMPLATE.md]: This should also ask whether they're using it in the browser or in node.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+    * :apple: `:apple:` when fixing something on macOS
+    * :checkered_flag: `:checkered_flag:` when fixing something on Windows
+    * :bug: `:bug:` when fixing a bug
+    * :fire: `:fire:` when removing code or files
+    * :green_heart: `:green_heart:` when fixing the CI build
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :shirt: `:shirt:` when removing linter warnings
```
Comment (willclarktech) [inline CONTRIBUTING.md]: I'm all for this, but we're not doing it ourselves. Is this something we definitely want?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
+# Contributing to Lisk-JS
+
+First off, thanks for taking the time to contribute! :raised_hands:
+
+The following is a set of guidelines for contributing to Lisk-JS, which are hosted in the [LiskHQ Organization](https://github.com/LiskHQ) on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+#### Table Of Contents
+
+[Code of Conduct](#code-of-conduct)
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Table Of Contents should be an ordered list, no?

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+This project and everyone participating in it is governed by the [Lisk-JS Code of Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [info@lisk.io](mailto:info@lisk.io).
+
+## Project License
+
+Every repository within LiskHQ comes with a LICENSE file. Please read it carefully before commiting your code to one of the repositories.
+
+
+## Help! I don't want to read this whole thing, I just have a question. :mag_right:
+
+Lisk is an open-source decentralized project, there are many ways and platforms to get help. These are some of them
```
Comment (willclarktech) [inline CONTRIBUTING.md]: End with `:`

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+
+* [Discuss and Ask on Reddit](https://www.reddit.com/r/Lisk/)
+* [Lisk FAQ](https://docs.lisk.io/docs/faq)
+
+If you prefer to chat with LiskHQ and other developers directly::
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Duplicate `:`

File: CONTRIBUTING.md (hunk 8 above)
Comment (willclarktech) [reply CONTRIBUTING.md]: Unsafe for what?

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+We will do our best to keep `master` in good shape, with tests passing at all times. But in order to move fast, we will make API changes that your application might not be compatible with. We will do our best to communicate these changes and always version appropriately so you can lock into a specific version if need be.
+
+### Pull Requests
+
+In case you've never submitted a pull request (PR) via GitHub before, please read [this short tutorial](https://help.github.com/articles/creating-a-pull-request). If you've submitted a PR before, there should be nothing surprising about our procedures for Lisk.
+
+*Before* submitting a pull request, please make sure the following is done:
+
+1. Fork the repo and create your branch from `master`.
+2. If you've added code that should be tested, add tests!
```
Comment (willclarktech) [inline CONTRIBUTING.md]: What code shouldn't be tested?

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+3. Ensure the test suite passes (`npm test`).
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Also linting.

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+4. Submit a pull request via GitHub.
+5. Check that Jenkins CI tests pass (pull request turns green).
+New developers probably need to wait for a trusted team member to start Jenkins CI on a Pull Request
```
Comment (willclarktech) [inline CONTRIBUTING.md]: `First time contributors will need to wait`

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+
+* Include issue numbers in the PR title, at the end with: ```Description - Closes #IssueNumber```
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Why is there a mix of ordered and unordered lists? Does the second list need some introduction or is it really part of the first list?

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* Follow the [JavaScript](https://github.com/airbnb/javascript) styleguide from airbnb with the [lisk extension](https://github.com/LiskHQ/eslint-config-lisk-base).
```
Comment (willclarktech) [inline CONTRIBUTING.md]: `from Airbnb`

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+* Document new code based on the [YuiDoc Documentation Styleguide](http://yui.github.io/yuidoc/)
+* End all files with a newline
+* Every new code must come with complete test coverage
+* Update the Readme for the changes that adhere to your new code
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Inconsistent full stops at the end of these list items.

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+2. If you've added code that should be tested, add tests!
+3. Ensure the test suite passes (`npm test`).
+4. Submit a pull request via GitHub.
+5. Check that Jenkins CI tests pass (pull request turns green).
+New developers probably need to wait for a trusted team member to start Jenkins CI on a Pull Request
+
+* Include issue numbers in the PR title, at the end with: ```Description - Closes #IssueNumber```
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* Follow the [JavaScript](https://github.com/airbnb/javascript) styleguide from airbnb with the [lisk extension](https://github.com/LiskHQ/eslint-config-lisk-base).
+* Document new code based on the [YuiDoc Documentation Styleguide](http://yui.github.io/yuidoc/)
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Do we want this in here since we'll be removing it soon?

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,162 @@
...
+* End all files with a newline
+* Every new code must come with complete test coverage
```
Comment (willclarktech) [inline CONTRIBUTING.md]: `All new code` - but this was covered above?

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+3. Ensure the test suite passes (`npm test`).
+4. Submit a pull request via GitHub.
+5. Check that Jenkins CI tests pass (pull request turns green).
+New developers probably need to wait for a trusted team member to start Jenkins CI on a Pull Request
+
+* Include issue numbers in the PR title, at the end with: ```Description - Closes #IssueNumber```
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* Follow the [JavaScript](https://github.com/airbnb/javascript) styleguide from airbnb with the [lisk extension](https://github.com/LiskHQ/eslint-config-lisk-base).
+* Document new code based on the [YuiDoc Documentation Styleguide](http://yui.github.io/yuidoc/)
+* End all files with a newline
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Generally, should conform to the `.editorconfig` rules

File: CONTRIBUTING.md (hunk 22 above)
Comment (willclarktech) [reply CONTRIBUTING.md]: We used all-caps for the `LICENSE` file above.

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,162 @@
...
+
+#### How Do I Submit A (Good) Bug Report?
+
+Bugs are tracked as [GitHub issues](https://guides.github.com/features/issues/). After you've determined [which repository](https://github.com/LiskHQ) your bug is related to, create an issue on that repository and provide the following information by filling in [the template](ISSUE_TEMPLATE.md).
+
+Explain the problem and include additional details to help maintainers reproduce the problem:
+
+* **Use a clear and descriptive title** for the issue to identify the problem.
+* **Describe the exact steps which reproduce the problem** in as many details as possible. For example, start by explaining how you started Lisk-JS, e.g. which command exactly you used in the terminal, or how you started Lisk-JS otherwise. When listing steps, **don't just say what you did, but explain how you did it**. For example, if you moved the cursor to the end of a line, explain if you used the mouse, or a keyboard shortcut or an Lisk-JS command, and if so which one?
```
Comment (willclarktech) [inline CONTRIBUTING.md]: The example here is specific to Atom. We should have one that's relevant to lisk-js

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,162 @@
...
+* If the problem started happening recently, **can you reproduce the problem in an older version of Lisk-JS?** What's the most recent version in which the problem doesn't happen? You can download older versions of Lisk-JS from [the releases page](https://github.com/LiskHQ/lisk-js/releases).
+* **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+Include details about your configuration and environment:
+
+* **Which version of Lisk / Lisk-JS are you using?**
+* **What's the name and version of the OS you're using**?
+* **Are you running Lisk-JS or Lisk in a virtual machine?** If so, which VM software are you using and which operating systems and versions are used for the host and the guest?
+* **Are you using [local configuration files](https://docs.lisk.io/docs)** `config.json`? If so, provide the contents of those files, preferably in a [code block](https://help.github.com/articles/markdown-basics/#multiple-lines) or with a link to a [gist](https://gist.github.com/).
```
Comment (willclarktech) [inline CONTRIBUTING.md]: We should probably have a line reminding people NEVER SHARE YOUR SECRET PASSPHRASES OR PRIVATE KEYS. Maybe in the "Describe the exact steps" section. Here we should probably remind people to hide sensitive config values.

File: CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,162 @@
...
+
+* **Use a clear and descriptive title** for the issue to identify the suggestion.
+* **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+* **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+* **Describe the current behavior** and **explain which behavior you expected to see instead** and why.
+* **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of Lisk-JS which the suggestion is related to. You can use [this tool](http://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux.
+* **Explain why this enhancement would be useful** to most Lisk and Lisk-JS users.
+* **List some other text editors or applications where this enhancement exists.**
```
Comment (willclarktech) [inline CONTRIBUTING.md]: We're not a text editor or an application.

File: CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,162 @@
...
+    * :bug: `:bug:` when fixing a bug
+    * :fire: `:fire:` when removing code or files
+    * :green_heart: `:green_heart:` when fixing the CI build
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :shirt: `:shirt:` when removing linter warnings
+
+### JavaScript Styleguide
```
Comment (willclarktech) [inline CONTRIBUTING.md]: We should probably mention we use `eslint`.

File: CONTRIBUTING.md (hunk 8 above)
Comment (Tosch110) [reply CONTRIBUTING.md]: Unsafe for using in production

File: CONTRIBUTING.md (hunk 23 above)
Comment (Tosch110) [reply CONTRIBUTING.md]: Yes, let us change this once we change it in the code and apply the new documentation tools.

File: CONTRIBUTING.md (hunk 13 above)
Comment (Tosch110) [reply CONTRIBUTING.md]: Since it states `Consider starting` I thought maybe we want to consider it, too

File: ISSUE_TEMPLATE.md [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+**Expected behavior:** [What you expect to happen]
+
+**Actual behavior:** [What actually happens]
+
+**Reproduces how often:** [What percentage of the time does it reproduce?]
+
+### Versions
+
+Please provide the version you are working with. Please include the OS and what version of the OS you're running.
+Also please provide if you are using the NodeJS or Browser version of Lisk-JS.
```
Comment (willclarktech) [inline ISSUE_TEMPLATE.md]: "Also please tell us whether you are using Lisk-JS in a browser (which one? which version?) or in Node.js."

File: CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,159 @@
...
+1. [Code of Conduct](#code-of-conduct)
+
+2. [Help! I don't want to read this whole thing, I just have one question. :mag_right:](#help!-i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
+
+3. [How Can I Contribute?](#how-can-i-contribute)
+    1. [Reporting Bugs](#reporting-bugs)
+    1. [Suggesting Enhancements](#suggesting-enhancements)
+    1. [Pull Requests](#pull-requests)
+
+4. [Styleguides](#styleguides)
```
Comment (willclarktech) [inline CONTRIBUTING.md]: Inconsistent numbering styles.

File: CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,159 @@
...
+### Pull Requests
+
+In case you've never submitted a pull request (PR) via GitHub before, please read [this short tutorial](https://help.github.com/articles/creating-a-pull-request). If you've submitted a PR before, there should be nothing surprising about our procedures for Lisk.
+
+*Before* submitting a pull request, please make sure the following is done:
+
+1. Fork the repo and create your branch from `master`.
+2. Add tests to the code you have contributed!
+All new code must come with complete test coverage.
+End all files with a newline. Generally your code should conform to the .editorconfig rules.
```
Comment (willclarktech) [inline CONTRIBUTING.md]: "In general, your code should conform to the rules listed in the `.editorconfig` file. There are [plugins for most editors/IDEs](http://editorconfig.org/#download) to do this for you automatically."

File: CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,159 @@
...
+Document new code based on the [YuiDoc Documentation Styleguide](http://yui.github.io/yuidoc/).
+Update the README for the changes that adhere to your new code.
+3. Ensure the test and linting suite passes (`npm run prepush`).
```
Comment (willclarktech) [inline CONTRIBUTING.md]: "`npm run prepush` runs both".

File: CONTRIBUTING.md [hunk 34]
```diff
@@ -0,0 +1,159 @@
...
+
+In case you've never submitted a pull request (PR) via GitHub before, please read [this short tutorial](https://help.github.com/articles/creating-a-pull-request). If you've submitted a PR before, there should be nothing surprising about our procedures for Lisk.
+
+*Before* submitting a pull request, please make sure the following is done:
+
+1. Fork the repo and create your branch from `master`.
+2. Add tests to the code you have contributed!
+All new code must come with complete test coverage.
+End all files with a newline. Generally your code should conform to the .editorconfig rules.
+Document new code based on the [YuiDoc Documentation Styleguide](http://yui.github.io/yuidoc/).
```
Comment (willclarktech) [inline CONTRIBUTING.md]: "Document new code with inline comments"

File: CONTRIBUTING.md [hunk 35]
```diff
@@ -0,0 +1,159 @@
...
+
+#### How Do I Submit A (Good) Bug Report?
+
+Bugs are tracked as [GitHub issues](https://guides.github.com/features/issues/). After you've determined [which repository](https://github.com/LiskHQ) your bug is related to, create an issue on that repository and provide the following information by filling in [the template](ISSUE_TEMPLATE.md).
+
+Explain the problem and include additional details to help maintainers reproduce the problem:
+
+* **Use a clear and descriptive title** for the issue to identify the problem.
+* **Describe the exact steps which reproduce the problem** in as many details as possible. For example, start by explaining how you started Lisk-JS, e.g. with NodeJS or in the Browser, or how you started Lisk-JS otherwise. When listing steps, **don't just say what you did, but explain how you did it**. For example, if you have used an API model provide the configuration you have chosen and the functions you have executed. **Make sure to erase sensitive information from the configuration or details you are passing - NEVER SHARE YOUR SECRET PASSPHRASES OR PRIVATE KEYS**.
```
Comment (willclarktech) [inline CONTRIBUTING.md]: "with Node.js or in a browser (which one? which version?)"

Comment (willclarktech) [reply CONTRIBUTING.md]: The bit about configuration overlaps with the section below on configuration and environment.

Comment (Tosch110) [reply CONTRIBUTING.md]: Yes, thanks. Configuration and environment are all mentioned in this block, the extra block for it has been taken out.

**Pull request merged**